eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -139,6 +145,60 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
         checkForMerges(submittedVariantEntities);
     }
 
+    /**
+     * This method handles a specific scenario in clustering where an already clustered variant is remapped and
+     * no matching rs id exist. In this case, clustered variant should continue using its existing rs id and an entry
+     * needs to be made ClusteredVariantEntity collection.
+     *
+     *                     SubmittedVariantEntity                                                           ClusteredVariantEntity
+     * SS	HASH	        RS	ASM	    STUDY	CONTIG	POS	REF	ALT                             RS	HASH	        ASM	    POS	CONTIG	TYPE
+     * 500	ASM1_Chr1_1_A_T	300	ASM1	PRJEB1	Chr1	1	A	T    (original)                 300	ASM1_Chr1_1_SNV	ASM1	1	Chr1	SNV
+     * 500	ASM2_Chr1_2_A_T	300	ASM2	PRJEB1	Chr1	2	A	T    (remapped to asm2)
+     *
+     * No RS id exist for the HASH ASM2_Chr1_2_A_T,	but it has an existing rs id of 300, which it should continue to use.
+     * It should not generate a new rs id for this situation as they will get merged right after generation and will be wasted
+     *
+     *         ClusteredVariantEntity (after operation)
+     *          RS	HASH	        ASM	    POS	CONTIG	TYPE
+     *          300	ASM1_Chr1_1_SNV	ASM1	1	Chr1	SNV
+     *          300	ASM2_Chr1_2_SNV	ASM1	1	Chr1	SNV
+     */
+    private List<IClusteredVariant> processClusteredVariantsWhereNoRSExists(List<? extends SubmittedVariantEntity> submittedVariantEntities) {
+        List<SubmittedVariantEntity> submittedVariantWithRS = submittedVariantEntities.stream()
+                .filter(v -> v.getClusteredVariantAccession() != null)
+                .collect(Collectors.toList());
+        List<? extends SubmittedVariantEntity> submittedVariantsNoRSExists = getSubmittedVariantsNoRSExists(submittedVariantWithRS);
+        List<ClusteredVariantEntity> clusteredVariantEntityNoRSExists = submittedVariantsNoRSExists.stream()
+                .map(this::toClusteredVariantEntity)
+                .collect(Collectors.toList());
+        mongoTemplate.insert(clusteredVariantEntityNoRSExists, ClusteredVariantEntity.class);
```

Comment:
This should only be done for variants that have been remapped so it would be better to validate that. 
Variants that are clustered and not remapped should always have a corresponding/existing RS id but still, to be safe I suggest adding the check.

Reply:
added a filter to take only remapped submittedVariants